op-service/sources/prefetching_eth_client.go

```diff
@@ -0,0 +1,192 @@
+package sources
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+type PrefetchingEthClient struct {
+	inner            *EthClient
+	prefetchingRange uint64
+}
+
+// NewPrefetchingEthClient creates a new [PrefetchingEthClient] with the given underlying [EthClient]
+// and a prefetching range.
+func NewPrefetchingEthClient(inner *EthClient, prefetchingRange uint64) (*PrefetchingEthClient, error) {
+	return &PrefetchingEthClient{
+		inner:            inner,
+		prefetchingRange: prefetchingRange,
+	}, nil
+}
+
+func (p *PrefetchingEthClient) FetchWindow(ctx context.Context, start, end uint64) {
+	for i := start; i <= end; i++ {
+		// Ignoring the error and result as this is just prefetching
+		// The actual fetching and error handling will be done when the data is requested
+		// FLAG FOR CODE REVIEW: Should we be creating new contexts here? The same context is just passed around everywhere.
+		go p.FetchBlockAndReceipts(ctx, i)
```

Comment:
You already start the `FetchWindow` calls in its own goroutines below, I would call these here just sequentially. Should also take off some load on the rpc service if a lot of requests are happening in parallel.
```suggestion
		p.FetchBlockAndReceipts(ctx, i)
```